desktop/extensions-sdk/architecture/index.md

```diff
@@ -0,0 +1,113 @@
+To configure and run those components, Docker Desktop uses a `metadata.json` file. See the
+[metadata](./metadata) section for more details.
+
+## The frontend
+
+The frontend, is basically a web application made from HTML, Javascript, and CSS. It can be built with a simple HTML
+file, some vanilla Javascript or any frontend framework, such as React or Vue.js.
+
+When Docker Desktop installs the extension, it extracts the UI folder from the extension image, as defined by the `ui` section in the `metadata.json`. See the [ui metadata section](./metadata.md#ui) for more details.
+
+Every time users click on the **Extensions** tab, Docker Desktop starts the UI. When they close the tab, the UI is
+not running anymore.
+
+The frontend can invoke `docker` commands, communicate with the extension backend, or invoke extension executables
+deployed on the host, through the [Extensions SDK](https://www.npmjs.com/package/@docker/extension-api-client){: target="_blank" rel="noopener" class="_"}.
+
+> **Tip**
+>
+> The `docker extension init` generates a React based extension. But you can still use it as a starting point for
+> your own extension and use any other frontend framework, like Vue, Angular, Svelte, etc. or event stay with
+> vanilla Javascript.
+{: .tip }
+
+Learn more about [building a frontend](../build/set-up/frontend-extension-tutorial.md) for your extension.
+
+## The backend
+
+Alongside a frontend application, extensions can also contain a backend service. In many cases, the Extension does not need a backend, and features can be implemented just by invoking docker commands through the SDK. However, there are some cases when an extension requires a backend
+	service, for example:
+- run long-running processes that must outlive the frontend
+- store data in a local database and serve them back with a REST API
+- store the extension state, like when a button starts a long-running process, so that if you navigate away
+  from the extension and come back, the frontend can pick up where it left off
+- accessing specific resources in the Docker Desktop VM, for example by mounting folders in the compose
+file
+
+> **Tip**
+>
+> The `docker extension init` generates a Go backend. But you can still use it as a starting point for
+> your own extension and use any other language like Node.js, Python, Java, .Net, or any other language and framework.
+{: .tip }
+
+Usually, the backend is made of one container that runs within the Docker Desktop VM. Internally, Docker Desktop creates
+a Docker Compose project, creates the container from the `image` option of the `vm` section of the `metadata.json`, and
+attaches it to the Compose project. See the [ui metadata section](./metadata.md#vm) for more details.
+
+In many situations, extension backend services can be defined by using the same image also used to package the extension.
+This image must then have a defined `CMD` to start the backend service, in addition to `COPY` lines for the extension packaging.
```

Comment:
I copied that from the content of the old metadata page.

I think that this sentence tries to explain how to package the extension so that the same image contains the UI files and the backend binaries, and once run, it runs the backend service.
How would you rephrase that to make it more straightforward?